gost94/src/gost94.rs

```diff
@@ -238,31 +237,30 @@ impl Update for Gost94 {
     }
 }
 
-impl FixedOutput for Gost94 {
+impl FixedOutputDirty for Gost94 {
     type OutputSize = U32;
 
-    fn finalize_fixed(mut self) -> GenericArray<u8, U32> {
-        {
-            let self_state = &mut self.state;
+    fn finalize_into_dirty(&mut self, out: &mut GenericArray<u8, U32>) {
```

Comment:
I think it could be worth to be consistent and use `digest::Output<Self>` across all crates.

Reply:
The reason I couldn't here is because it doesn't `impl Default`, and `digest::Output` (presently, at least) uses the `Digest` trait:

https://github.com/RustCrypto/traits/blob/master/digest/src/digest.rs#L96

It could potentially use `FixedOuput` instead.

Reply:
Hrmm, changing it to use `FixedOutput` is not trivial... it means you can't use it with `Digest` because not all `Digest`s necessarily impl `FixedOutput`, and that kind of defeats the point.

I think we should leave it as-is. That said, I think we can fix the version used in the macro.